### PR TITLE
Added polynomial free energy and material to set the phase field parameters

### LINKED ISSUE #5076: Create polynomial free energy material
I will implement the polynomial free energies using a material class with automatic differentiation.

### PR DESCRIPTION
**Pull request by tonkmr**
I added the PolynomialFreeEnergy material using the expression builder system. I also added a material that calculates the phase field parameters for a quantitative polynomial free energy model. Closes #5076

### CODE REVIEW

File: modules/phase_field/src/materials/PFParamsPolyFreeEnergy.C [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+  Real surf_energy = _surface_energy*_JtoeV*_length_scale*_length_scale;
+
+  // Set interfacial parameter and energy barrier
+  _kappa[_qp] = surf_energy*_int_width/KN;
+  _W[_qp] = surf_energy/( 2.0*_int_width*KN );
+
+  Real Co = 0.0;
+  Real a = _c_eq[_qp];
+
+  switch (_order)
```
Comment (dschwen) [inline modules/phase_field/src/materials/PFParamsPolyFreeEnergy.C]: Is there a reason you don't fold those two switch statements into one?

**Comment by dschwen** [comment]
The rest looks good... except for a few OCD issues I have with the indentation of the line continuation of the constructor parameter lists and the comment style for the class documentation in the PFParamsPolyFreeEnergy header.

File: modules/phase_field/src/materials/PFParamsPolyFreeEnergy.C (hunk 1 above)
Comment (tonkmr) [inline modules/phase_field/src/materials/PFParamsPolyFreeEnergy.C]: I liked keeping the calculation of KN and Co separate, to keep the code easier to understand. It makes it easy to differentiate the various options for KN and the various options for Co.

**Comment by jwpeterson** [comment]
I'll fix @dschwen's OCD issues and merge.

**Pull request merged**